### PR TITLE
closes #153

### LINKED ISSUE #153: Support SQS metric for oldest message in queue
There is a new Cloudwatch metric available now `ApproximateAgeOfOldestMessage` which was announced here: https://aws.amazon.com/about-aws/whats-new/2016/08/new-amazon-cloudwatch-metric-for-amazon-sqs-monitors-the-age-of-the-oldest-message

Have not looked too much into it just thought I would get it on the radar.

### PR DESCRIPTION
**Pull request by majormoses**
## Pull Request Checklist

**Is this in reference to an existing issue?**
#153 

#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [x] Update README with any necessary configuration snippets

- [x] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass 

#### Purpose

Add support for checking metrics such as ApproximateAgeOfOldestMessage. Updated sdk changes while I was in it.

#### Known Compatablity Issues
unknown

- Added support for checking different metric types
- Updated to latest aws sdk

### CODE REVIEW

File: bin/check-sqs-messages.rb [hunk 1]
```diff
@@ -121,9 +128,8 @@ def run
       crit = false
       queues = []
 
-      sqs.queues.with_prefix(config[:prefix]).each do |q|
-        queue_name = q.arn.split(':').last
-        messages = q.approximate_number_of_messages
+      sqs.queues(queue_name_prefix: config[:prefix]).each do |q|
+        messages = sqs.client.get_queue_attributes(queue_url: q.url, attribute_names: ['All']).attributes[config[:metric]].to_i
```
Comment (eheydrick) [inline bin/check-sqs-messages.rb]: I get `Check failed to run: undefined local variable or method queue_name` when using a queue prefix.  `queue_name = q.attributes['QueueArn'].split(':').last` would do the trick.

Comment (majormoses) [inline bin/check-sqs-messages.rb]: @eheydrick can you give me your example command you are running as I am not seeing that locally when I test:

Testing full queue name:
```
$ bundle exec ./bin/check-sqs-messages.rb -r us-west-2 -q dev-aws-cleaner -w 1000 -c 1200 -m ApproximateAgeOfOldestMessage
SQSMsgs OK: 0 messages in dev-aws-cleaner queue
```

Testing prefix
```
$ bundle exec ./bin/check-sqs-messages.rb -r us-west-2 -p dev -w 1000 -c 1200 -m ApproximateAgeOfOldestMessage
SQSMsgs OK: All queues matching prefix 'dev' OK
```

**Comment by majormoses** [comment]
@eheydrick updated to use common as requested, need a little more information on replicating what you are seeing.

**Comment by eheydrick** [comment]
@majormoses if you set a condition that causes the check to return a warn or crit you can see the error. If I add the `queue_name` variable back per above it works as expected.

`$ /opt/sensu/embedded/bin/ruby ./check-sqs-messages.rb -r us-west-2 -p dev -W 100`

```Check failed to run: undefined local variable or method `queue_name' for #<SQSMsgs:0x00000000e4de50>
Did you mean?  queues, ["./check-sqs-messages.rb:140:in `block in run'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.27/lib/aws-sdk-resources/batch.rb:75:in `each'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.27/lib/aws-sdk-resources/batch.rb:75:in `each'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.27/lib/aws-sdk-resources/batch.rb:75:in `each'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.27/lib/aws-sdk-resources/collection.rb:18:in `block in each'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.27/lib/aws-sdk-resources/operations.rb:142:in `block in all_batches'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.6.27/lib/aws-sdk-core/pageable_response.rb:75:in `each'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.27/lib/aws-sdk-resources/operations.rb:141:in `all_batches'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.27/lib/aws-sdk-resources/collection.rb:18:in `each'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/aws-sdk-resources-2.6.27/lib/aws-sdk-resources/collection.rb:18:in `each'", "./check-sqs-messages.rb:131:in `run'", "/opt/sensu/embedded/lib/ruby/gems/2.3.0/gems/sensu-plugin-1.4.3/lib/sensu-plugin/cli.rb:58:in `block in <class:CLI>'"]```

**Comment by majormoses** [comment]
@eheydrick perfect was able to replicate and put in a fix. Here it is working now:
```
$ bundle exec ./bin/check-sqs-messages.rb -r us-west-2 -q dev-pipeline-data-manager-priority_deadletter -W 100
SQSMsgs WARNING: 0 message(s) in dev-pipeline-data-manager-priority_deadletter queue
```

```
$ bundle exec ./bin/check-sqs-messages.rb -r us-west-2 -p dev -W 100
SQSMsgs WARNING: ["0 message(s) in <REDACTED>, <TRUNCATED>]
```

**Comment by eheydrick** [comment]
Looks 👍 Thanks @majormoses!

**Pull request merged**